Gopkg.toml

```diff
@@ -0,0 +1,90 @@
+
+# Gopkg.toml example
+#
+# Refer to https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
+#
+# [[constraint]]
+#   name = "github.com/user/project"
+#   version = "1.0.0"
+#
+# [[constraint]]
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#  name = "github.com/x/y"
+#  version = "2.4.0"
+
+
+[[constraint]]
+  name = "github.com/Masterminds/semver"
+  version = "1.3.1"
+
+[[constraint]]
+  name = "github.com/Masterminds/sprig"
+  version = "2.12.0"
+
+[[constraint]]
+  name = "github.com/sirupsen/logrus"
+  version = "1.0.3"
+
+[[constraint]]
+  name = "github.com/cheggaaa/pb"
+  version = "1.0.18"
+
+[[constraint]]
+  branch = "master"
```

Comment:
If I read the [constraints documentation on Gopkg.toml](https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md#constraint) correctly, I believe our hierarchy for defining version should be the following:

- Version, when available.
- Revision, if a suitable version is not available
- Branch, never.

If I understand the behavior of dep correctly, if we rebuild these dependencies we will have unintended code changes if we are pinned just to a branch here. We definitely don't want that.

Reply:
I don't think this dep should change unless we were to run an `dep ensure -update`. Provided the hash in the lock file is valid, I believe we should still get the same version checked out unless the hash is not valid for some reason.

That said, I'm not sure we gain anything by defining a constraint for these packages if we're defining master as the constraint. Perhaps we should remove them?

Reply:
Ahh, yes. I see that now. According to the [updating dependencies](https://github.com/golang/dep#updating-dependencies) section of the readme even running `dep ensure -update` is not recommended, as you would instead run it on an individual package, such as `dep ensure -update github.com/drud/go-pantheon`.

With that in mind, I'm probably fine with the branches. I'd still prefer versions when available, but it doesn't seem like this will cause the instability I was concerned about.

Reply:
i approve this line of thinking 😄